### ISSUE TITLE
Form view is goes to top when ever i click on the button or If any user error raise

### ISSUE THREAD

**Issue by SaravananSakthivel** [odoo/odoo#46060]
Impacted versions: 12.0 CE

Steps to reproduce: 

1. Place a button inside the form where you can scroll down the access so that you can achieve this error (ex.purchase order -> action_set_date_planned)
2. Whenever we click on the button, form is going to top in edit mode.

Current behavior:

1. Whenever we click on the button, form is going to top

Expected behavior:

1. Whenever we click on the button, the form should stay in the place where it is actually existing

Video/Screenshot link (optional):
![ezgif com-crop](https://user-images.githubusercontent.com/3658948/75145397-e1f84100-571e-11ea-9234-7d8d6a3ea246.gif)

**Comment by pedrobaeza** [comment]
I don't consider that as an error, maybe an improvement. The explanation why this happens is because the form is reloaded, and thus, it returns to top.

@Yenthe666 what is your opinion?

**Comment by SaravananSakthivel** [comment]
@pedrobaeza yes it is not an error. I am trying to say about the behavior. I have a 50-70 fields in my customized form(Where users enters data in every day basis) and there is a button to fetch some data in the data entry flow. Users gets frustrated of scrolling into the same form.

It must be improved to work as a user friendly.
Note: It is not going to top when it is not in edit mode.

**Comment by pedrobaeza** [comment]
Well, I personally think that a design with 50-70 fields all together is not a good design, but that's my personal opinion. Any way, you have several workarounds:

* Put the button on top, not on bottom.
* Split information across pages of a notebook and put the button on the header.

Any way, don't expect this to be changed in the current version. Maybe it can land as an improvement in master (future version).

**Comment by SaravananSakthivel** [comment]
@pedrobaeza  In some cases we have to go with customer requirement.
You know core account invoice has more than 50 fields.

As of now we are placed button in header only. But as a user friendly button can be placed anywhere 
Hope it will get sorted out soon in next or current version.

**Comment by Yenthe666** [comment]
Hi,

I agree with @pedrobaeza that this is not a bug and that having 50-70 fields on one form to fill in is not a good design. Maybe your design needs some functional rethinking or more logic to automatically fill in values to work with?

That being said, I think we should add this as an improvement for master. It would be great if the button clicks keep track of the height of the form :) @ged-odoo any feedback from your end on this?

**Comment by SaravananSakthivel** [comment]
As You guys suggesting about design part, please note that all these fields are not entered manually.There are some calculated fields.As a user entry it will be around 20 fields. Indian company with production scenario, there are some fields which has to be captured(from client point).May be you are not come through with this kind of scenario.

I have raised the issue from the improvement side of the frame work.

**Comment by ged-odoo** [comment]
This is indeed not really a bug, but this is an improvement that would make sense. This is actually a quite simple change (there are `getLocalState` and `setLocalState` methods implemented in the form renderer that could simply export and restore the scroll position.

**Comment by ged-odoo** [comment]
@aab-odoo

**Comment by ged-odoo** [comment]
@ged-odoo

**Comment by aab-odoo** [comment]
@ged-odoo

**Comment by nilshamerlinck** [comment]
For the record, I faced a similar issue and in my case the position change was due to the autofocus mechanism, that was focusing on the first `default_focus` element (required field).

2 options:
- set `default_focus="0"` on said field
- or disabling autofocus on the whole form, like [here](https://github.com/odoo/odoo/commit/7297aa03195e6161a7b43833ee592da4a19ff159)

@ged-odoo could this useful [small change](https://github.com/odoo/odoo/commit/7297aa03195e6161a7b43833ee592da4a19ff159#diff-1d14645ab67f799283cfda5ef40553d9a0a37d5b9f25f4c15f9d08aca91eff42R41) be backported to 12.0/13.0?

**Comment by albertopoljak** [comment]
Any updates on this?
